### PR TITLE
wallet+db: fix SQL-wallet review issues

### PR DESCRIPTION
**Pull request by yyforyongyu**
Addresses review findings and regressions surfaced while preparing the
SQL-wallet/runtime-store stack. This is the base branch for #1268.

Highlights:
- Fix taproot PSBT decoration so wallet-decorated taproot inputs do not carry
  both legacy and taproot derivations.
- Keep missing kvdb scopes consistent with `Store.ListAccounts` empty-list
  semantics.
- Snapshot recovery outpoints before same-block lookahead expansion.
- Harden SQL block consistency checks and sparse rollback sync-height rewinds.
- Surface kvdb watch-only wallet state and guard wallet account validation
  against nil extended pubkeys.
- Accept value-typed scoped account and coin-source inputs.
- Update the rollback corruption assertion to check backend-independent error
  context.

Validation:
- `make fmt`
- `make lint`
- `scripts/check-each-commit.sh sql-wallet`
- Focused wallet, kvdb, and DB integration regression tests

### CODE REVIEW

File: wallet/internal/db/kvdb/accountstore_listaccounts.go [hunk 1]
```diff
@@ -198,6 +198,13 @@ func selectScopedManagers(mgr waddrmgr.AddrStore, filter *db.KeyScope) (
 	}
 
 	scopedMgr, err := mgr.FetchScopedKeyManager(scope)
+	if waddrmgr.IsError(err, waddrmgr.ErrScopeNotFound) {
```
Comment (yyforyongyu) [inline wallet/internal/db/kvdb/accountstore_listaccounts.go:201]: This flips `ListAccounts` missing-scope behavior from `ErrAccountNotFound` to an empty result but adds no regression test, and no existing test exercises `ListAccounts` with an unregistered scope (the "unknown scope" cases cover only the Create paths) — add a test asserting a missing-scope query returns an empty result with a nil error so the fix and its parity with the SQL backend stay locked in.

(`claude-opus-4-8[1m]`)

**Comment by yyforyongyu** [comment]
No issues found by gpt-5.5 🧊

File: wallet/internal/db/kvdb/accountstore_listaccounts.go [hunk 2]
```diff
@@ -198,6 +198,13 @@ func selectScopedManagers(mgr waddrmgr.AddrStore, filter *db.KeyScope) (
...
+		// List queries return an empty result for an unknown scope,
+		// matching the Store contract and the SQL backend, rather than
+		// surfacing ErrAccountNotFound.
+		return nil, nil
+	}
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply wallet/internal/db/kvdb/accountstore_listaccounts.go:206]: Addressed — `TestListAccountsMissingScopeReturnsEmpty` (`wallet/internal/db/kvdb/accountstore_test.go`) exercises `ListAccounts` with an unregistered key scope and asserts an empty result with a nil error, locking in parity with the SQL backends and the documented `AccountStore.ListAccounts` contract.

File: wallet/internal/db/kvdb/accountstore_listaccounts.go (hunk 1 above)
Comment (yyforyongyu) [reply wallet/internal/db/kvdb/accountstore_listaccounts.go:201]: Addressed — `TestListAccountsMissingScopeReturnsEmpty` (`wallet/internal/db/kvdb/accountstore_test.go`) exercises `ListAccounts` with an unregistered key scope and asserts an empty result with a nil error, locking in parity with the SQL backends and the documented `AccountStore.ListAccounts` contract.

File: wallet/recovery.go [hunk 3]
```diff
@@ -328,7 +328,28 @@ func (rs *RecoveryState) ProcessBlock(block *wire.MsgBlock) (
 		foundHorizons   map[waddrmgr.BranchScope]uint32
 	)
 
+	// filterTx mutates rs.outpoints in place: it deletes a watched outpoint
+	// the moment it sees a spend of it. When a same-block lookahead hit
```
Comment (yyforyongyu) [inline wallet/recovery.go]: Q: when will a same-block lookahead be triggered?

File: wallet/recovery.go [hunk 4]
```diff
@@ -328,7 +328,28 @@ func (rs *RecoveryState) ProcessBlock(block *wire.MsgBlock) (
...
+	// triggers a horizon expansion we re-run filterBlock over the same
+	// block, but by then the spent outpoint is already gone, so the second
+	// pass no longer recognizes that spend and the overwritten final result
+	// would silently drop a spend-only transaction. Snapshot the watched
+	// set up front and restore it before every pass so each filterBlock run
+	// observes the same starting outpoints and is idempotent; the final
+	// pass then leaves rs.outpoints with the correct post-block state.
+	outpointsSnapshot := make(map[wire.OutPoint][]byte, len(rs.outpoints))
+	for op, script := range rs.outpoints {
```
Comment (yyforyongyu) [inline wallet/recovery.go]: turn this into a helper method? sth like `copyOutpointMap`?

**Comment by yyforyongyu** [comment]
No issues found by `gpt-5.5 xhigh` 🦉

**Comment by yyforyongyu** [comment]
No issues found by `claude-opus-4-8[1m]` 🎯

File: wallet/recovery.go (hunk 3 above)
Comment (yyforyongyu) [reply wallet/recovery.go]: Addressed by clarifying the recovery comment. A same-block lookahead is triggered when processing an output extends an account horizon; the same block may contain a later output to one of the newly derived addresses, so recovery reruns detection over the block with the expanded horizon before moving on.

File: wallet/recovery.go (hunk 4 above)
Comment (yyforyongyu) [reply wallet/recovery.go]: Addressed by adding `copyOutpointMap` and using it for the spend snapshot before same-block expansion.

**Comment by yyforyongyu** [comment]
No issues found by `claude-opus-4-8[1m]` 🧹

**Comment by yyforyongyu** [comment]
No issues found by `gpt-5.5 xhigh` 🦝

**Review by GustavoStingelin — APPROVED** [review]
LGTM! 🛡️
